managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2182,21 +2187,12 @@ public boolean hasMoreEntries(PositionImpl position) {
         return result;
     }
 
-    void discardEntriesFromCache(ManagedCursorImpl cursor, PositionImpl newPosition) {
-        Pair<PositionImpl, PositionImpl> pair = activeCursors.cursorUpdated(cursor, newPosition);
-        if (pair != null) {
-            entryCache.invalidateEntries(pair.getRight());
-        }
-    }
-
     public PositionImpl getEvictionPosition(){
         PositionImpl evictionPos;
         if (config.isCacheEvictionByMarkDeletedPosition()) {
-            PositionImpl earlierMarkDeletedPosition = getEarlierMarkDeletedPositionForActiveCursors();
-            evictionPos = earlierMarkDeletedPosition != null ? earlierMarkDeletedPosition.getNext() : null;
+            evictionPos = activeCursors.getSlowestReaderPosition();
```

Comment:
Miss offset?

Reply:
@nodece activeCursors contains the active cursors ordered by mark delete position. What did you mean with your question?

Reply:
the tests in ManagedLedgerTest also verify the behavior

Reply:
Old behavior is `earlierMarkDeletedPosition.getNext()`, new behavior without `getNext()`.

Reply:
the tests confirm that it's correct now. I believe that it was invalid previously.